pkg/daemon/metrics.go

```diff
@@ -186,6 +316,57 @@ func extractRegularMetrics(processName, output string) (offsetFromMaster, maxOff
 	return
 }
 
+// updateClockStateMetrics ...
+func updateClockStateMetrics(process, iface string, state string) {
+	if state == LOCKED {
+		ClockState.With(prometheus.Labels{
+			"process": process, "node": NodeName, "iface": iface}).Set(1)
+	} else {
+		ClockState.With(prometheus.Labels{
+			"process": process, "node": NodeName, "iface": iface}).Set(0)
+	}
+}
+
+func UpdateInterfaceRoleMetrics(process string, iface string, role ptpPortRole) {
+	InterfaceRole.With(prometheus.Labels{
+		"process": process, "node": NodeName, "iface": iface}).Set(float64(role))
+}
+
+func extractPTP4lEventState(output string) (portId int, role ptpPortRole) {
+	replacer := strings.NewReplacer("[", " ", "]", " ", ":", " ")
+	output = replacer.Replace(output)
+
+	index := strings.Index(output, " port ")
+	if index == -1 {
+		return
+	}
+	output = output[index:]
+	fields := strings.Fields(output)
+	portIndex := fields[1]
+	role = UNKNOWN
+	var e error
+	portId, e = strconv.Atoi(portIndex)
+	if e != nil {
+		glog.Errorf("error parsing port id %s", e)
+		portId = 0
+		return
+	}
+
+	if strings.Contains(output, "UNCALIBRATED to SLAVE") {
+		role = SLAVE
+	} else if strings.Contains(output, "UNCALIBRATED to PASSIVE") || strings.Contains(output, "MASTER to PASSIVE") ||
+		strings.Contains(output, "SLAVE to PASSIVE") {
+		role = PASSIVE
+	} else if strings.Contains(output, "UNCALIBRATED to MASTER") || strings.Contains(output, "LISTENING to MASTER") {
+		role = MASTER
+	} else if strings.Contains(output, "FAULT_DETECTED") || strings.Contains(output, "SYNCHRONIZATION_FAULT") {
+		role = FAULTY
+	} else {
+		portId = 0
```

Comment:
should here by role = unknown?

Reply:
role is already set to UNKNOWN  before if condition as a default value